### ISSUE TITLE
Native Module cannot be null

### ISSUE THREAD

**Issue by trevans24** [joonhocho/react-native-google-sign-in#46]
I have followed the directions for both IOS and Android. When I simply try to import it via:
`import GoogleSignIn from 'react-native-google-sign-in';`
react native throws this error: Requiring module "react-native-google-sign-in/js/GoogleSignIn.ios.js" which threw an exception: Invariant Violation: Native module cannot be null.
no idea why this is happening or why this is not working with an import, cannot find any documentation to this error before or if anyone else is having this issue as well.

**Comment by sravanb** [comment]
even I am getting the same error. 

<img width="616" alt="screen shot 2017-08-19 at 6 30 41 pm" src="https://user-images.githubusercontent.com/512833/29486821-8b7dda10-850c-11e7-9599-b9e2a2bd573e.png">

**Comment by idlework** [comment]
Same

**Comment by idlework** [comment]
@trevans24 and @sravanb 

To solve this issue you should add the contents of the ```{YourApp}/node_modules/react-native-google-sign-in/ios/RNGoogleSignIn``` folder to the same group where your ```appDelegate.m``` is located. Then make sure in these files are added to the ```Build phases -> Compile sources``` list.

**Comment by rajeshkos** [comment]
Can you please explain this ? @trevans24

**Comment by joonhocho** [comment]
@trevans24 let me know if this is resolved.

**Comment by john1jan** [comment]
I am also getting the same error

![screen shot 2017-09-14 at 7 18 15 pm](https://user-images.githubusercontent.com/4264813/30433273-ab95ffc4-9981-11e7-8824-f0ba194a8873.png)

**Comment by john1jan** [comment]
Issue is in GoogleSignIn.ios.js
```
const {
  RNGoogleSignIn,
  RNGoogleSignInEvents,
} = NativeModules;
```
RNGoogleSignIn and RNGoogleSignInEvents are undefined

**Comment by john1jan** [comment]
When i open my ios xcodeproject and try to run the app.
It says Google/SignIn.h file not found

![screen shot 2017-09-14 at 7 31 14 pm](https://user-images.githubusercontent.com/4264813/30433917-65667518-9983-11e7-875d-d4b48555fccb.png)

**Comment by john1jan** [comment]
I think this problem is  because I havn't setup swift bridging header.
 And the [link](http://www.learnswiftonline.com/getting-started/adding-swift-bridging-header/) given in the readme to setup doc swift bridging header is not working

**Comment by diegomelo182** [comment]
Same issue.

**Comment by john1jan** [comment]
I was able to setup google sign in using [react-native-google-signin](https://github.com/devfd/react-native-google-signin). It has good documentation for setting up on IOS

**Comment by spencewine** [comment]
For anyone having this problem pull requests #52  and #54 resolved this issue for me.

**Comment by joonhocho** [comment]
@trevans24 @sravanb @idlework @rajeshkos @john1jan @diegomelo182 @spencewine 
Just upgraded to the latest Google SignIn SDK for iOS. (npm version 1.0.16)
Please follow new instructions carefully.
https://github.com/joonhocho/react-native-google-sign-in#ios
Also refer to ExampleApp if something is not working. I've tested ExampleApp and is working fine.
Feel free to reopen this issue if you cannot get it working.

**Comment by gesf** [comment]
By having the content of `/node_modules/react-native-google-sign-in/ios/RNGoogleSignIn` added to the project I get a Swift version issue when building. My Xcode version is 9.2 and the available Swift versions are 3.2/4. Any idea on how to solve this ? (if I don't add the above folder and get the mentioned error).

**Comment by skycloud112** [comment]
The ExampleApp does not work for me:

![image](https://user-images.githubusercontent.com/7694000/37546362-d1428552-2942-11e8-8d3f-d92f718dcd59.png)

I did npm install and ran react-native run-ios, the error is: Print: Entry, ":CFBundleIdentifier", Does Not Exist
